applications/parameter_sweep.md

```diff
@@ -0,0 +1,229 @@
+        return tuple(l for l in viewer.layers if isinstance(l, layer_type))
+    except AttributeError:
+        return ()
+
+def show_layer_result(gui, result, return_type) -> None:
+    if result is None:
+        return
+    try:
+        viewer = gui.parent().qt_viewer.viewer
+    except AttributeError:
+        return
+    try:
+        viewer.layers[gui.result_name].data = result
+    except KeyError:
+        adder = getattr(viewer, f"add_{return_type.__name__.lower()}")
+        adder(data=result, name=gui.result_name)
+
+register_type(layers.Layer, choices=get_layers, return_callback=show_layer_result)
+
+```
+
+We then create a viewer with napari, and add the `data.page()` image layer to try out parameter sweep with different thresholding techniques. We use the `magicgui` decorator for the method `apply_threshold` to turn it into a magicgui. 
+
+`auto_call = True` indicates magicgui to call `apply_threshold` whenever the value of a parameter changes. The QDoubleSlider is used to pass the size parameter to the function, and we offer the choices to threshold an image layer using the adaptive, Sauvola and Niblack techniques:
+
+```python
+with gui_qt():
+    # creating a viewer and adding the data.page() image layer
+    viewer = Viewer()
+    viewer.add_image(data.page(), name="page")
+    
+    # passing size as (2* floor(size/2)+1), as block-size and window size have to be odd natural numbers
+    @magicgui(
+        auto_call=True,
+        size={"widget_type": QDoubleSlider, "maximum": 99, "fixedWidth": 400},
+        technique={"choices": ["adaptive", "niblack", "sauvola"]},
+    )
+    def apply_threshold(layer: layers.Image, size=1, technique="adaptive") -> None:
+        if layer:
+            if technique=="adaptive":
+                return layer.data > threshold_local(layer.data, (2* floor(size/2)+1))
+            elif technique=="sauvola":
+                return layer.data > threshold_sauvola(layer.data, (2* floor(size/2)+1))
+            elif technique=="niblack":
+                return layer.data > threshold_niblack(layer.data, (2* floor(size/2)+1))
+  ```
+
+Then, we instantiate our magicgui widget and connect it to the current viewer for updation of layer drop-down choices. We also add the gui as a widget to our viewer:
+
+```python          
+    gui = apply_threshold.Gui()
+    gui.parentChanged.connect(gui.refresh_choices)
+    viewer.layers.events.changed.connect(lambda x: gui.refresh_choices("layer"))
+    viewer.window.add_dock_widget(gui)
+```
+
+The `show_result` method is a callback function to add the thresholded image to the layers. This updates the resultant layer whenever the `apply_threshold` method is called:
+```python
+
+    def show_result(result):
+          try:
+            viewer.layers["thresholded"].data = result
+        except KeyError:
+            viewer.add_image(data=result, name="thresholded")
+
+       apply_threshold.called.connect(show_result)
+ ```
```

Comment:
```suggestion
```

this can be removed if you add the `-> layers.Image:` type annotation to the function